src/lib/reducer.ts

```diff
@@ -19,62 +19,44 @@ export type Action =
         id: number;
       };
     }
-  | {
-      type: "uncheck";
-      payload: {
-        id: number;
-      };
-    }
   | {
       type: "delete_task";
     };
 
 export const reducer = (state: State, action: Action): State => {
   const task_data = _.cloneDeep(state.task);
-  const comp_data = _.cloneDeep(state.comp);
 
   switch (action.type) {
     case "genelate_task":
-      if (action.payload.data.data !== "") {
-        return {
-          task: [...state.task, action.payload.data],
-          comp: [...state.comp],
-        };
-      } else {
-        return {
-          task: [...state.task],
-          comp: [...state.comp],
-        };
-      }
-
+      return {
+        nextId: state.nextId + 1,
+        task: [
+          ...state.task,
+          { id: state.nextId, data: action.payload.taskName, flag: false },
+        ],
+      };
     case "check":
       for (let i = 0; i < state.task.length; i++) {
         if (state.task[i].id === action.payload.id) {
-          task_data.splice(i, 1);
-          comp_data.push(state.task[i]);
+          task_data[i].flag = !task_data[i].flag;
+
           break;
         }
       }
 

```

Comment:
同じ処理やけど、for文は`of`を使うともう少し楽かな。
あとは、せっかくディープコピーしたんやけ、`state.task`は使わなくていいと思う。

```suggestion
       for (const task of task_data) {
        if (task.id === action.payload.id) {
          task.flag = !task.flag;

          break;
        }
      }
```